components/clp-package-utils/clp_package_utils/controller.py

```
@@ -0,0 +1,762 @@
+    user_id: int,
+    group_id: int,
+) -> None:
+    """
+    Recursively changes the owner of the given path to the given user ID and group ID.
+
+    :param path:
+    :param user_id:
+    :param group_id:
+    """
+    chown_cmd = ["chown", "--recursive", f"{user_id}:{group_id}", str(path)]
+    subprocess.run(chown_cmd, stdout=subprocess.DEVNULL, check=True)
```

Ruff (0.14.0)

`subprocess` call: check for execution of untrusted input

(S603)